crates/npmrc/src/custom_deserializer.rs

```diff
@@ -131,18 +140,26 @@ mod tests {
     #[test]
     fn test_default_store_dir_with_xdg_env() {
         env::set_var("XDG_DATA_HOME", "/tmp/xdg_data_home");
-        let home_dir = dirs::home_dir().unwrap();
         let store_dir = default_store_dir();
-        assert_eq!(store_dir, home_dir.join("/tmp/xdg_data_home/pacquet/store"));
+        assert_eq!(store_dir, PathBuf::from("/tmp/xdg_data_home/pacquet/store"));
         env::remove_var("XDG_DATA_HOME");
     }
 
     #[test]
-    fn test_default_store_dir_with_xdg_env_with_prefix() {
-        env::set_var("XDG_DATA_HOME", "~/prefix/xdg_data_home");
+    fn test_is_home() {
+        let with_prefix = is_home("~/.pacuqet".to_string());
+        assert_eq!(with_prefix, true);
+        let without_prefix = is_home("/tmp/store".to_string());
+        assert_eq!(without_prefix, false);
+    }
+
+    #[test]
+    fn test_resolve_store_dir() {
         let home_dir = dirs::home_dir().unwrap();
-        let store_dir = default_store_dir();
-        assert_eq!(store_dir, home_dir.join("prefix/xdg_data_home/pacquet/store"));
-        env::remove_var("XDG_DATA_HOME");
+        let store_dir = resolve_store_dir("~/.store".to_string());
+        assert_eq!(store_dir, home_dir.join(".store"));
+
+        let store_dir_without_prefix = resolve_store_dir("/tmp/store".to_string());
+        assert_eq!(store_dir_without_prefix, PathBuf::from("/tmp/store"))
     }
```

Comment:
If this test fails on Windows, please copy the test and set different `cfg` (`#[cfg(unix)]` for Linux and macOS, and `#[cfg(windows)` for Windows).